assets/js/components/Menu.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Menu component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import useMergedRef from '@react-hook/merged-ref';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useCallback, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { MDCMenu } from '../material-components';
+
+const Menu = forwardRef( ( {
+	menuOpen,
+	menuItems,
+	onSelected,
+	id,
+}, ref ) => {
+	let menu = null;
+	let menuEl = null;
```

Comment:
@aaemnnosttv I reverted to `useRef` but also introduced `useState` to store the menu so that we don't need to instantiate the menu each time. Let me know if this approach looks good to you?